src/verifiable/keys.ts

```diff
@@ -126,3 +126,46 @@ function getX5cChain(
 export function isX5C(params: BasicParameters): params is KeyParameters {
   return 'x5c' in params
 }
+
+/**
+ * The jose library has 2 types of instances: JWK<Type>Key and JWK.<TypeKey>. The former is just data interface with no
+ * methods where as the latter is a richer type. This method converts from the data interface to the richer type.
+ *
+ * @param jwk - The instance to convert.
+ * @returns The converted result.
+ */
+export function toKey(
+  jwk: JWKRSAKey | JWKECKey | JWKOctKey | JWKOKPKey,
+): JWK.RSAKey | JWK.ECKey | JWK.OKPKey | JWK.OctKey {
+  if (jwk.kty === 'EC') {
+    return JWK.asKey(jwk)
+  }
+  if (jwk.kty === 'oct') {
+    return JWK.asKey(jwk)
+  }
+  if (jwk.kty === 'OKP') {
+    return JWK.asKey(jwk)
+  }
+  return JWK.asKey(jwk)
+}
```

Comment:
Do we need the `if` statements here?

This looks like we're only calling `JWK.asKey(jwk)`.

Reply:
Typescript is doing some magical type inference based on these assertions in order to target the correct method. It wouldn't compile without doing it this way. I believe it's related to conditional types: https://medium.com/javascript-everyday/conditional-types-in-typescript-770ac9a64f46

Reply:
Oh TypeScript. Worth a code comment?